MSCL/source/mscl/MicroStrain/MIP/Packets/MipSharedDataFields.h

```diff
@@ -0,0 +1,134 @@
+        //  Note: returns true if qualifier is CH_UNKNOWN and there is only one data point for the specified field
+        bool has(MipTypes::ChannelField field, MipTypes::ChannelQualifier qualifier = MipTypes::CH_UNKNOWN) const;
+
+    public:
+        //API Constructor: MipSharedDataFields
+        //  Create an empty MipSharedDataFields object
+        MipSharedDataFields() {};
+
+        //API Function: hasEventSource
+        //  Checks whether the packet contains the Event Source field
+        //
+        //Returns:
+        //  bool - true if Event Source is present
+        bool hasEventSource() const;
+
+        //API Function: eventSource
+        //  Gets the Event Source value for the packet
+        //
+        //Returns:
+        //  uint8 - the event source trigger ID, 0x00 if not triggered by event, 0xFF if Event Source not present
+        uint8 eventSource() const;
+
+        //API Function: hasTicks
+        //  Checks whether the packet contains the Ticks field
+        //
+        //Returns:
+        //  bool - true if Ticks is present
+        bool hasTicks() const;
+
+        //API Function: ticks
+        //  Gets the Ticks value for the packet
+        //
+        //Returns:
+        //  uint32 - the ticks value
+        uint32 ticks() const;
+
+        //API Function: hasDeltaTicks
+        //  Checks whether the packet contains the Delta Ticks field
+        //
+        //Returns:
+        //  bool - true if Delta Ticks is present
+        bool hasDeltaTicks() const;
+
+        //API Function: deltaTicks
+        //  Gets the Delta Ticks value for the packet
+        //
+        //Returns:
+        //  uint32 - the delta ticks value
+        uint32 deltaTicks() const;
+
+        //API Function: hasGpsTimestamp
+        //  Checks whether the packet contains the GPS Timestamp field
+        //
+        //Returns:
+        //  bool - true if GPS Timestamp is present
+        bool hasGpsTimestamp() const;
+
+        //API Function: gpsTimestamp
+        //  Gets the GPS Timestamp value for the packet.
+        //  Note: On Gen 7 products and later this will generally be the same value as MipDataPacket::deviceTimestamp()
+        //
+        //Returns:
+        //  <Timestamp> - the GPS timestamp value
+        Timestamp gpsTimestamp() const;
+
+        //API Function: hasDeltaTime
+        //  Checks whether the packet contains the Delta Time field
+        //
+        //Returns:
+        //  bool - true if Delta Time is present
+        bool hasDeltaTime() const;
+
+        //API Function: deltaTime
+        //  Gets the Delta Time value for the packet
+        //
+        //Returns:
+        //  double - the delta time value
+        double deltaTime() const;
+
+        //API Function: get
+        //  Get the shared field <MipDataPoint> with the specified field and qualifier.
+        //  If qualifier is CH_UNKNOWN and there is only one data point for the field, that field will be returned.
```

Comment:
Can link a reference to CH_UNKNOWN in docs with `<MipTypes::CH_UNKNOWN>`
Can be changed for other references like this, but not critical of course